Support for `pointer_to_address` instruction [ISSUE# 81]

taufique71 opened the pull request:
<!-- What's in this pull request? -->
Changes to handle `pointer_to_address` instruction

<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->
Resolves [Issue 81](https://github.com/themaplelab/swift/issues/81).

taufique71 committed:
Implemented pointer_to_address instruction

karimhamdanali commented on lib/WALASupport/SILWalaInstructionVisitor.cpp:
```diff
@@ -1421,6 +1421,23 @@ jobject SILWalaInstructionVisitor::visitAddressToPointerInst(AddressToPointerIns
 }
 
+jobject SILWalaInstructionVisitor::visitPointerToAddressInst(PointerToAddressInst *PTAI) {
+
```
remove this empty line

mmroz commented:
So, I looked at the two instructions address_to_pointer and pointer_to_address and these 2 instructions are conversion instructions even though the types are internal. In the PR https://github.com/themaplelab/swift/pull/138 I made that change address_to_pointer and I think we should do that here too! Just add:

  jobject ConvertedTypeNode = Wala->makeConstant(CovertedType.c_str());
  jobject RawPointerCastNode = Wala->makeNode(CAstWrapper::CAST, ConvertedNode, ConvertedTypeNode);

karimhamdanali commented:
In addition to @mmroz comments, please resolve the merge conflicts before I approving this PR.

taufique71 committed:
Modification to PR #137 according to review

- Removed blank like at the start of the function
- Created new CAST node to wrap the value node and represent the
  conversion
- Added some comments on the implementation [Not in the review]

taufique71 commented:
@karimhamdanali, everything resolved. Please take a look if it's okay now.

karimhamdanali commented:
Thanks @taufique71. @mmroz, do these changes address your concerns as well?

mmroz commented:
Looks good!
